game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/retreat/OffensiveGeneralRetreat.java

```diff
@@ -0,0 +1,129 @@
+package games.strategy.triplea.delegate.battle.steps.retreat;
+
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.ATTACKER_WITHDRAW;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.MustFightBattle.RetreatType;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import games.strategy.triplea.delegate.battle.steps.RetreatChecks;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import javax.annotation.Nullable;
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import org.triplea.java.collections.CollectionUtils;
+
+@AllArgsConstructor
+public class OffensiveGeneralRetreat implements BattleStep {
+
+  final BattleState battleState;
+
+  final BattleActions battleActions;
+
+  @Override
+  public List<String> getNames() {
+    if (canAttackerRetreat()
```

Comment:
Nitpicky - Even if a multi-line, I still think this would be cleaner as a ternary. I think with context, it seems easier, but I would suggest that maintaining/reading the code, it's easier to grok as a ternary. Perhaps the boolean can be extracted to a helper method to make it a simple tenary. WDYT?